edc-extensions/ssi/ssi-miw-credential-client/src/main/java/org/eclipse/tractusx/edc/iam/ssi/miw/api/MiwApiClientImpl.java

```diff
@@ -170,12 +170,20 @@ private <R> Result<R> handleSuccess(Response response, TypeReference<R> tr) {
     }
 
     private <R> Result<R> handleError(Response response) {
-        var msg = format("MIW API returned %s", response.code());
+        var body = "";
+        if (response.body() != null) {
+            try {
+                body = response.body().string();
+            } catch (IOException e) {
+                monitor.debug("Failed to read response from MIW");
+                return Result.failure(e.getMessage());
+            }
+        }
+        var msg = format("MIW API returned %s with body: %s", response.code(), body);
```

Comment:
We should take care to not log any sensitive data in terms MIW returns some in message body.

Reply:
It should not i guess/hope 
but in case we can also not forward the body to the Result and just use that on the debug logging